File: go/raindrops/raindrops.go
// Package raindrops provides Raindrops solution.
package raindrops

import (
	"strconv"
)

// Convert converts given number into raindrop sounds.
func Convert(n int) (sounds string) {
	if n%3 == 0 {
		sounds += "Pling"
	}
	if n%5 == 0 {
		sounds += "Plang"
	}
	if n%7 == 0 {
		sounds += "Plong"
	}
	if len(sounds) == 0 {
		return strconv.Itoa(n)
	}
	return sounds
}

File: fsharp/raindrops/Raindrops.fs
module Raindrops

let convert (number: int): string =
    [
        if number % 3 = 0 then Some "Pling"
        if number % 5 = 0 then Some "Plang"
        if number % 7 = 0 then Some "Plong"
    ]
    |> List.choose id
    |> function
        | [] -> string number
        | xs -> String.concat "" xs
